example/cajita_tutorial/09_grid_parallel/grid_parallel_example.cpp

```diff
@@ -0,0 +1,207 @@
+    std::cout << "Total grid sum: " << sum << " (should be "
+              << 4 * ghost_is.size() << ")" << std::endl;
+
+    /*
+      Grid parallel operations are also possible by directly using an index
+      space instead of using the index space from within the local grid. Here we
+      subtract 2 from every entity.
+    */
+    auto own_is = local_grid->indexSpace( Cajita::Own(), Cajita::Cell(),
+                                          Cajita::Local() );
+    Cajita::grid_parallel_for(
+        "index_space_for", exec_space(), own_is,
+        KOKKOS_LAMBDA( const int i, const int j, const int k ) {
+            for ( int l = 0; l < 4; ++l )
+                array_view( i, j, k, l ) -= 2.0;
+        } );
+
+    /*
+      Just as before we can do a grid reduction as well.
+    */
+    double sum_is = 0.0;
+    Cajita::grid_parallel_reduce(
+        "index_space_reduce", exec_space(), own_is,
+        KOKKOS_LAMBDA( const int i, const int j, const int k, double& result ) {
+            for ( int l = 0; l < 4; ++l )
+                result += array_view( i, j, k, l );
+        },
+        sum_is );
+    /*
+      Now the total should be the negative of the total owned grid size, again
+      times the four field values for each grid point.
+    */
+    std::cout << "Total grid sum: " << sum_is << " (should be "
+              << -4 * own_is.size() << ")" << std::endl;
+
+    /*
+      One potentially useful extension of directly using index spaces is that
+      they can be fused together into a single parallel kernel. Here we iterate
+      over both the local and boundary index spaces at the same time, combined
+      in a Kokkos::Array and differentiated with the index in the array in the
+      kernel.
+    */
+    auto boundary_is = local_grid->boundaryIndexSpace(
+        Cajita::Own(), Cajita::Cell(), 1, 0, 0 );
+    Cajita::grid_parallel_for(
+        "multi_space_for", exec_space{},
+        Kokkos::Array<Cajita::IndexSpace<3>, 2>{ boundary_is, own_is },
+        KOKKOS_LAMBDA( const int s, const int i, const int j, const int k ) {
+            for ( int l = 0; l < 4; ++l )
+            {
+                if ( 0 == s )
+                    array_view( i, j, k, l ) = 5.0;
+                else if ( 1 == s )
+                    array_view( i, j, k, l ) = 3.0;
+            }
+        } );
+    /*
+      Here we reduce only over each index space separately to check the value.
+    */
+    double sum_bound = 0.0;
+    Cajita::grid_parallel_reduce(
+        "boundary_space_reduce", exec_space(), boundary_is,
+        KOKKOS_LAMBDA( const int i, const int j, const int k, double& result ) {
+            for ( int l = 0; l < 4; ++l )
+                result += array_view( i, j, k, l );
+        },
+        sum_bound );
+    /*
+      This total should be the size of the boundary, times the value, times the
+      four field values for each grid point.
+    */
+    std::cout << "Low-X-boundary grid sum: " << sum_bound << " (should be "
```

Comment:
I'm doing something wrong here - I think I misunderstand something about iterating over multiple index spaces

Reply:
The result should be additive? The index spaces are iterated over sequentially

Reply:
Yeah I was overwriting the values, thanks